include/sound/sof/ext_manifest4.h

```diff
@@ -0,0 +1,122 @@
+/* SPDX-License-Identifier: ((GPL-2.0 WITH Linux-syscall-note) OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+/*
+ * Extended manifest is a place to store metadata about firmware, known during
+ * compilation time - for example firmware version or used compiler.
+ * Given information are read on host side before firmware startup.
+ * This part of output binary is not signed.
+ * Note that not all items in the struct are 32 bits length, and we assume
+ * the firmware is little-endian.
+ */
+
+#ifndef __SOF_FIRMWARE_EXT_MANIFEST4_H__
+#define __SOF_FIRMWARE_EXT_MANIFEST4_H__
+
+#include <linux/uuid.h>
+
+/* In ASCII  $AE1 */
+#define SOF_MAN4_MAGIC_NUMBER 0x31454124
+
+#define SOF_MAN4_MAX_MODULE_NAME_LEN 8
+#define SOF_MAN4_MAX_FW_BINARY_NAME 8
+#define SOF_MAN4_DEFAULT_HASH_SHA256_LEN 32
+#define SOF_MAN4_SEGMENT_NUM 3
+#define SOF_MAN4_MODULE_PARAMS_NUM 4
+#define SOF_MAN4_FW_HDR_OFFSET 0x2000
```

Comment:
@bardliao thanks but you didnt use the struct defined in the first patch?

Reply:
> @bardliao thanks but you didnt use the struct defined in the first patch?

Yes, I did. See [snd_sof_fw_ext_man4_parse](https://github.com/thesofproject/linux/pull/3176/commits/41951dfd31585a5ca5d216bedc8d89ab99ab7555#diff-a4132048dde4b0a1f8b761545f01a09bb681f6ecdf49d0d7bb374084ae4843aaR363)
```
static int snd_sof_fw_ext_man4_parse(struct snd_sof_dev *sdev, const struct firmware *fw)
{
	struct snd_sof_pdata *sof_pdata = sdev->pdata;
	const struct sof_dev_desc *desc = sof_pdata->desc;
	const struct snd_sof_ext_man_pdata *ext_man_pdata = desc->ext_man_pdata;
...
	if (ext_man_pdata)
		fw_hdr_offset = ext_man_pdata->fw_hdr_offset;
	else
		fw_hdr_offset = SOF_MAN4_FW_HDR_OFFSET;
...
}
```